libnetwork/iptables/iptables.go

```diff
@@ -88,54 +88,40 @@ func (e ChainError) Error() string {
 	return fmt.Sprintf("Error iptables %s: %s", e.Chain, string(e.Output))
 }
 
-func probe() {
-	path, err := exec.LookPath("iptables")
-	if err != nil {
-		logrus.Warnf("Failed to find iptables: %v", err)
-		return
-	}
-	if out, err := exec.Command(path, "--wait", "-t", "nat", "-L", "-n").CombinedOutput(); err != nil {
-		logrus.Warnf("Running iptables --wait -t nat -L -n failed with message: `%s`, error: %v", strings.TrimSpace(string(out)), err)
-	}
-	_, err = exec.LookPath("ip6tables")
-	if err != nil {
-		logrus.Warnf("Failed to find ip6tables: %v", err)
-		return
-	}
-}
-
-func initFirewalld() {
-	if err := FirewalldInit(); err != nil {
-		logrus.Debugf("Fail to initialize firewalld: %v, using raw iptables instead", err)
-	}
-}
-
 func detectIptables() {
 	path, err := exec.LookPath("iptables")
 	if err != nil {
+		logrus.Warnf("Failed to find iptables: %v", err)
 		return
 	}
 	iptablesPath = path
 
-	supportsXlock = exec.Command(iptablesPath, "--wait", "-L", "-n").Run() == nil
+	supportsXlock = true
+	if out, err := exec.Command(path, "--wait", "-L", "-n").CombinedOutput(); err != nil {
+		logrus.Warnf("Running iptables --wait -L -n failed with message: `%s`, error: %v", strings.TrimSpace(string(out)), err)
+	}
```

Comment:
> This looks to be changing the behavior

Ah right, `supportsXlock` is not turned on at the right place.

> While we're changing this, it may be good to use the WithError() approach for logging the error (we're trying to converge our logging to use that)

That's good to know, I'll try to remember that for future PRs. Is there a place where this sort of improvements are listed (if there's any other)?